src/main/java/Parser/Parser.java

```diff
@@ -0,0 +1,111 @@
+package Parser;
+
+import Command.ByeCommand;
+import Command.ClearCommand;
+import Command.Command;
+import Command.ListCommand;
+import Command.MarkCommand;
+import Command.UnmarkCommand;
+import Command.TodoCommand;
+import Command.DeleteCommand;
+import Command.DeadlineCommand;
+import Command.EventCommand;
+import Command.FindCommand;
+import DukeException.DukeException;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * A parser that takes the input data and checks its syntax followed bt excuting it.
+ */
+public class Parser {
+
+    final String UNKNOWN_COMMAND = "☹ OOPS!!! I'm sorry, but I don't know what that means :-(";
+
+    /**
+     * A method that takes the user input and calls the right command for it.
+     *
+     * @param fullCommand the user command input.
+     * @return a command that will be excuted base on the users input.
+     * @throws DukeException if the command is invalid in any ways.
+     */
+    public static Command parse(String fullCommand) throws DukeException{
+        String[] split = fullCommand.split(" ", 2);
+        String commandAction = split[0];
+        switch (commandAction) {
+            case "list":
```

Comment:
switch case should be inline with no indents for case clauses as per java coding standard